service/dap/server.go

```diff
@@ -1389,30 +1389,34 @@ func (s *Server) onThreadsRequest(request *dap.ThreadsRequest) {
 
 		if next >= 0 {
 			s.logToConsole(fmt.Sprintf("Too many goroutines, only loaded %d", len(gs)))
-
-			// Make sure the selected goroutine is included in the list of threads
-			// to return.
-			if state != nil && state.SelectedGoroutine != nil {
-				var selectedFound bool
-				for _, g := range gs {
-					if g.ID == state.SelectedGoroutine.ID {
-						selectedFound = true
-						break
-					}
+		}
+		// Make sure the selected goroutine is included in the list of threads
+		// to return.
+		selectedIdx := -1
```

Comment:
For a moment I thought this was the goid. Maybe use -100 or constant IndexNotFound, so this is not confused with -1 for selected goroutine?